lib/travis/build/script/dart.rb

```diff
@@ -230,11 +230,10 @@ def dartanalyzer
           def dartfmt
             args = task[:dartfmt]
 
-            if args.is_a?(String)
-              sh.echo "dartfmt arguments aren't supported.", ansi: :red
-            end
-
-            sh.if package_direct_dependency?('dart_style'), raw: true do
+            # If specified `-dartfmt: custom` and there is a dependency on `dart_style` we
+            # will use the custom (pinned) version of dart_style to run formatting checks
+            # instead of the SDK.
+            sh.if args == 'custom' and package_direct_dependency?('dart_style'), raw: true do
```

Comment:
This doesn't work. `sh.if` takes a String as parameter and generates a shell if. Here `args == 'custom'` should rather be used alone with a ruby `if`.

Something like that should work:

```ruby
if args == 'custom'
  sh.if package_direct_dependency?('dart_style'), raw: true do
    sh.raw 'function dartfmt() { pub run dart_style:format "$@"; }'
  end
  sh.else do
    sh.failure "As 'custom' is used as dartfmt argument you should directly depend on dart_style in pubspec.yaml"
  end
elsif args.is_a?(String)
  sh.echo "dartfmt only support 'custom' as optional argument.", ansi: :red
end
```